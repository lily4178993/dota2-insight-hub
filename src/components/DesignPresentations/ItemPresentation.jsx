import React from 'react';
import PropTypes from 'prop-types';
import './itemPresentation.css';

function ItemPresentation({ data }) {
  if (!data || data.length === 0) {
    return (
      <div className="error-fetch">
        Something went wrong!
        <br />
        Item&apos;s data not found
      </div>
    );
  }
  return (
    <div className="item-presentation">
      <div className="item-details">
        <h2>Overview</h2>
        <p>
          <strong>Name:</strong>
          {' '}
          {data.dname}
        </p>
        <p>
          <strong>Type:</strong>
          {' '}
          {data?.qual}
        </p>
        <p>
          <strong>Mana Cost:</strong>
          {' '}
          {data?.mc ? 'Yes' : 'No'}
        </p>
        <p>
          <strong>Cost:</strong>
          {' '}
          {data?.cost}
          {' '}
          Gold
        </p>
        <p>
          <strong>Cooldown:</strong>
          {' '}
          {data?.cd ? `${data?.cd} seconds` : 'N/A'}
        </p>
        <div className="item-image">
          <strong>Item Image:</strong>
          <br />
          <img src={`https://api.opendota.com${data?.img}`} alt={data?.dname} />
        </div>
        <p>
          <strong>Lore:</strong>
          {' "'}
          {data?.lore}
          {'" '}
        </p>
        <p>
          <strong>Usage:</strong>
          {' '}
          {data?.hint[0]}
        </p>
        <h3>Special Notes:</h3>
        <p>{data?.notes}</p>
      </div>
    </div>
  );
}
ItemPresentation.propTypes = {
  data: PropTypes.shape({
    img: PropTypes.string.isRequired,
    dname: PropTypes.string.isRequired,
    qual: PropTypes.string.isRequired,
    lore: PropTypes.string.isRequired,
    notes: PropTypes.string.isRequired,
    cost: PropTypes.number.isRequired,
    mc: PropTypes.bool.isRequired,
    cd: PropTypes.number,
    hint: PropTypes.arrayOf(PropTypes.string).isRequired,
    length: PropTypes.number,
  }).isRequired,
};

export default ItemPresentation;
