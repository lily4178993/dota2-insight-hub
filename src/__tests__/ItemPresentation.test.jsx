import React from 'react';
import { render } from '@testing-library/react';
import ItemPresentation from '../components/DesignPresentations/ItemPresentation';

const sampleData = {
  img: '/sample-image.png',
  dname: 'Sample Item',
  qual: 'Sample Quality',
  lore: 'Sample Lore',
  notes: 'Sample Notes',
  cost: 100,
  mc: true,
  cd: 30,
  hint: ['Hint 1', 'Hint 2'],
};

test('ItemPresentation component matches snapshot', () => {
  const { asFragment } = render(<ItemPresentation data={sampleData} />);
  expect(asFragment()).toMatchSnapshot();
});
