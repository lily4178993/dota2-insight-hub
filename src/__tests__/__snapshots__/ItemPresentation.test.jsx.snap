// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPresentation component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="item-presentation"
  >
    <div
      class="item-details"
    >
      <h2>
        Overview
      </h2>
      <p>
        <strong>
          Name:
        </strong>
         Sample Item
      </p>
      <p>
        <strong>
          Type:
        </strong>
         Sample Quality
      </p>
      <p>
        <strong>
          Mana Cost:
        </strong>
         Yes
      </p>
      <p>
        <strong>
          Cost:
        </strong>
         100 Gold
      </p>
      <p>
        <strong>
          Cooldown:
        </strong>
         30 seconds
      </p>
      <div
        class="item-image"
      >
        <strong>
          Item Image:
        </strong>
        <br />
        <img
          alt="Sample Item"
          src="https://api.opendota.com/sample-image.png"
        />
      </div>
      <p>
        <strong>
          Lore:
        </strong>
         "Sample Lore" 
      </p>
      <p>
        <strong>
          Usage:
        </strong>
         Hint 1
      </p>
      <h3>
        Special Notes:
      </h3>
      <p>
        Sample Notes
      </p>
    </div>
  </div>
</DocumentFragment>
`;
